mex/common/utils.py

```diff
@@ -36,23 +44,100 @@ def any_contains_any(bases: Iterable[Container[T] | None], tokens: Iterable[T])
     return False
 
 
+def contains_only_types(field: "GenericFieldInfo", *types: type) -> bool:
+    """Return whether a `field` is annotated as one of the given `types`.
+
+    Unions, lists and type annotations are checked for their inner types and only the
+    non-`NoneType` types are considered for the type-check.
+
+    Args:
+        field: A `GenericFieldInfo` instance
+        types: Types to look for in the field's annotation
+
+    Returns:
+        Whether the field contains any of the given types
+    """
+    if inner_types := list(get_inner_types(field.annotation, include_none=False)):
```

Comment:
[i found one usage with out list()](https://github.com/robert-koch-institut/mex-common/blob/feature/mx-1702-field-and-type-utils/mex/common/models/base/model.py#L102) - but in general, i know that we wanted the generatores to fade out, but for this recursive case i actually find it way more readable to yield than to append and extend lists everywhere. what do you think?

Reply:
Yeah, so I'm skeptical about recursions in productive code to begin with. If it can't be avoided, I would prefer to abtract it away into an isolated flatten function and then do a regular iteration. But I don't want to open the can of worms of fundamental debate of recursions. So no hard objections. Leave it as it is.